Sources/SwiftProtobuf/Errors.swift

```diff
@@ -52,6 +56,8 @@ public enum DecodingError: Error {
     case malformedProtobuf
     /// Field names were not compiled into the binary
     case missingFieldNames
+    /// The field identifier (name or number) was not recognized
+    case unknownField
     /// TODO: More here?
```

Comment:
Note:  Were we to separate them, it would probably be good to split this into separate enums:  `ProtobufDecodingErrors`, `JSONDecodingErrors`, etc.  That way, switch exhaustiveness can be leveraged to be sure you've covered every possible error.